source/common/http/filter/ext_authz.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/config/filter/http/ext_authz/v2/ext_authz.pb.h"
+#include "envoy/ext_authz/ext_authz.h"
+#include "envoy/http/filter.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/common/assert.h"
+#include "common/ext_authz/ext_authz_impl.h"
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Http {
+namespace ExtAuthz {
+
+/**
+ * Type of requests the filter should apply to.
+ */
+enum class FilterRequestType { Internal, External, Both };
+
+/**
+ * Global configuration for the HTTP authorization (ext_authz) filter.
+ */
+class FilterConfig {
+public:
+  // TBD(saumoh) Take care of grpc service != envoy_grpc()
+  FilterConfig(const envoy::config::filter::http::ext_authz::v2::ExtAuthz& config,
+               const LocalInfo::LocalInfo& local_info, Stats::Scope& scope,
+               Runtime::Loader& runtime, Upstream::ClusterManager& cm)
+      : local_info_(local_info), scope_(scope), runtime_(runtime), cm_(cm),
+        cluster_name_(config.grpc_service().envoy_grpc().cluster_name()),
+        failure_mode_allow_(config.failure_mode_allow()) {}
+
+  const LocalInfo::LocalInfo& localInfo() const { return local_info_; }
+  Runtime::Loader& runtime() { return runtime_; }
+  Stats::Scope& scope() { return scope_; }
+  std::string cluster() { return cluster_name_; }
+  Upstream::ClusterManager& cm() { return cm_; }
+  bool failOpen() const { return failure_mode_allow_; }
```

Comment:
Please rename failOpen to failureModeAllow to avoid confusion.  While reviewing, I had to think about whether this was a plumbing or electrical circuit analogy, and ended up going to read the data-plane-api documentation.  I think using the original name makes it clearer.